test/e2e/call.e2e.js

```diff
@@ -0,0 +1,78 @@
+const { expect } = require('../test-setup');
+const { runForAtLeast } = require('../__lib__/mocha-utils');
+const cli = require('../__lib__/cli');
+const {
+	DEVICE_NAME,
+	PATH_PROJ_STROBY_INO
+} = require('../__lib__/env');
+
+
+describe('Call Command [@device]', () => {
+	const help = [
+		'Call a particular function on a device',
+		'Usage: particle call [options] <device> <function> [argument]',
+		'',
+		'Global Options:',
+		'  -v, --verbose  Increases how much logging to display  [count]',
+		'  -q, --quiet    Decreases how much logging to display  [count]',
+		'',
+		'Examples:',
+		'  particle call coffee brew                 Call the brew function on the coffee device',
+		'  particle call board digitalWrite D7=HIGH  Call the digitalWrite function with argument D7=HIGH on the board device'
+	];
+
+	before(runForAtLeast(25, async () => {
+		await cli.logout();
+		await cli.setProfile('e2e');
+		await cli.login();
+		await cli.flashTestFirmwareOTA(DEVICE_NAME, PATH_PROJ_STROBY_INO);
+	}));
+
+	after(async () => {
+		await cli.logout();
+		await cli.setDefaultProfile();
+	});
+
+	it('Shows `help` content', async () => {
+		const { stdout, stderr, exitCode } = await cli.run(['help', 'call']);
+
+		expect(stdout).to.equal('');
+		expect(stderr.split('\n')).to.include.members(help);
+		expect(exitCode).to.equal(0);
+	});
+
+	it('Shows `help` content when run without arguments', async () => {
+		const { stdout, stderr, exitCode } = await cli.run('call');
+
+		expect(stdout).to.equal('Parameter \'device\' is required.');
+		expect(stderr.split('\n')).to.include.members(help);
+		expect(exitCode).to.equal(1);
+	});
+
+	it('Shows `help` content when run with `--help` argument', async () => {
+		const { stdout, stderr, exitCode } = await cli.run(['call', '--help']);
+
+		expect(stdout).to.equal('');
+		expect(stderr.split('\n')).to.include.members(help);
+		expect(exitCode).to.equal(0);
+	});
+
+	it('Calls a function', async () => {
+		const args = ['call', DEVICE_NAME, 'check'];
+		const { stdout, stderr, exitCode } = await cli.run(args);
+
+		expect(stdout).to.equal('200');
+		expect(stderr).to.equal('');
+		expect(exitCode).to.equal(0);
+	});
+
+	it('Fails when attempting to call an unknown function', async () => {
+		const args = ['call', DEVICE_NAME, 'WATNOPE'];
+		const { stdout, stderr, exitCode } = await cli.run(args);
+
+		expect(stdout).to.equal('Function call failed: Function WATNOPE not found');
+		expect(stderr).to.equal('');
+		expect(exitCode).to.equal(1);
+	});
+});
```

Comment:
probably worth using mocha’s retry support though depending on which test fails, not all suites are retryable 🤨

Reply:
played around with this a bit. we'd need to move more of the setup / teardown logic into `beforeEach()` and `afterEach()` hooks but that would mean _lots_ of useless work and associated slowness. punting for now :+1: